packages/eslint-plugin/src/rules/promise-function-async.ts

```diff
@@ -37,6 +37,8 @@ export default createRule<Options, MessageIds>({
     fixable: 'code',
     messages: {
       missingAsync: 'Functions that return promises must be async.',
+      missingAsyncHybridReturn:
+        'Functions that return promises must be async. Consider adding an explicit return type annotation if the function is intended to return both a promise and a non-promise.',
```

Comment:
[Proofreading] I found this a big ambiguous, like a function is returning two things at once?

```suggestion
        'Functions that return promises must be async. Consider adding an explicit return type annotation if the function is intended to return either a promise or a non-promise.',
```

Reply:
Oh, definitely! Thanks! 🙂